src/fparser/two/utils.py

```diff
@@ -564,7 +570,13 @@ def match(startcls, subclasses, endcls, reader,
                 start_name = obj.get_start_name()
 
         # Comments and Include statements are always valid sub-classes
-        classes = subclasses + [Comment, Include_Stmt]
+        # Preprocessor directives are always valid sub-classes
+        classes = subclasses + [Comment, Include_Stmt, Cpp_If_Stmt,
```

Comment:
Could the class list from C99*.py be imported to provide the list of cpp classes?

Reply:
I was in fact using `get_module_classes` from parser.py for this in an earlier version and was under the impression that this slowed down the parser significantly (as the list is re-built quite often). I'll have a look again.

Reply:
I have now put a list with the relevant class names in C99Preprocessor.py. This list then used here as well as in `match_cpp_directive`. This way, the overhead seems to be non measurable.